docs/content/manual/manual.md

```diff
@@ -0,0 +1,5538 @@
+
+When given a JSON object as input, `.[k]` produces the value
+at the key `k` if it is present in the object, or `null` otherwise.
+For example, given the object `{name: "Anna", age: 24}`, the filter
+`.["name"]` produces `"Anna"`,
+`.["age"]` produces `24`, and
+`.["address"]` produces `null`.
+
+For [identifier-like keys](#objects) like `"foo"`, you can also look up
+the field `"foo"` of an object using the shorthand syntax `.foo`.
+For example, we could have written `.name`, `.age`, and `.address` above, whereas
+we cannot use this shorthand syntax for `.["foo::bar"]` and `.["foo.bar"]`.
+
+::: Compatibility
+In jq, when given `null` input, `.["a"]` and `.[0]` yield `null`, but `.[]` yields an error.
+jaq yields an error in all cases to prevent accidental indexing of `null` values.
+To obtain the same behaviour in jq and jaq, you can use
+`.["a"]? // null` or `.[0]? // null` instead.
+:::
+
+::: Examples
+
+~~~
+.foo
+{"foo": 42, "bar": "less interesting data"}
+42
+~~~
+
+~~~
+.foo
+{"notfoo": true, "alsonotfoo": false}
+null
+~~~
+
+~~~
+.["foo"]
+{"foo": 42}
+42
+~~~
+
+~~~
+.[0]
+[{"name":"JSON", "good":true}, {"name":"XML", "good":false}]
+{"name":"JSON", "good":true}
+~~~
+
+~~~
+.[2]
+[{"name":"JSON", "good":true}, {"name":"XML", "good":false}]
+null
+~~~
+
+~~~
+.[-2]
+[1,2,3]
+2
+~~~
+
+~~~
+.["a","b"]
+{"a":1, "b":2}
+1
+2
+~~~
+
+~~~
+.[keys[] | select(test("^b"))]
+{"a":1, "b":2, "c":3}
+2
+~~~
+
+:::
+
+## Slicing operator: `.[f:g]` {#slicing-operator}
+
+The operator `.[f:g]` returns a slice of an array or a string.
+For example, when given an array,
+`.[10:15]` returns an array of length 5, containing the
+elements from index 10 (inclusive) to index 15 (exclusive).
+Either index may be negative, in which case it counts
+backwards from the end of the array.
+If `f` is omitted, it is assumed to be `0`, and
+if `g` is omitted, it is assumed to be `length`.
+Indices are zero-based.
+
+::: Examples
+
+~~~
+.[2:4]
+["a","b","c","d","e"]
+["c", "d"]
+~~~
+
+~~~
+.[2:4]
+"abcdefghi"
+"cd"
+~~~
+
+~~~
+.[:3]
+["a","b","c","d","e"]
+["a", "b", "c"]
+~~~
+
+~~~
+.[-2:]
+["a","b","c","d","e"]
+["d", "e"]
+~~~
+
+~~~
+.[:range(1; length+1)]
+[1,2,3]
+[1]
+[1,2]
+[1,2,3]
+~~~
+
+:::
+
+## Compound paths
+
+Frequently, when using the path operators given above,
+we find ourselves combining them with the
+[`|`](#composition) and
+[`?`](#error-suppression) operators.
+Therefore, jq provides shorthand syntax for these combinations.
+For example:
+
+* `.key[]` for `.key | .[]`,
+* `.[].key?` for `.[] | .key?`,
+* `.[]?[]` for `.[]? | .[]`,
+* `.a.b` for `.a | .b`,
+* `.a.b.c` for `.a | .b | .c`, and so on.
+
+We call such a combination a _compound path_.
+
+The rules for what constitutes a compound path are surprisingly complex.
+Therefore, we define it via a formal grammar in
+[EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form):
+
+~~~ ebnf
+path = atomic, part
+     | ".", ident
+     | path, part
+     | path, part, "?"
```

Comment:
How can you parse `.foo??` in this definition?

Reply:
`.foo?` is a path, and `.foo??` is equivalent to `(.foo?)?`; that is, catching all errors from the path `.foo?`. We can see this by the following:

~~~
$ jq -n '.[error]?'
jq: error (at <unknown>) (not a string): null
$ jq -n '.[error]??'
$ jq -n '(.[error]?)?
~~~

(Here, the last two commands do not output anything, because the second `?` silences all errors of the path, including the error from `error`.)

Reply:
A path should be followed by `part` before `?` in your definition, but `.foo??` cannot be parsed because `part` is missing. If any path can be followed by `?`, it should be `path = pat "?"`?